From 77dc3caab301de14c8a54e152e7576b70a405673 Mon Sep 17 00:00:00 2001
From: Wendy Liang <wendy.liang@xilinx.com>
Date: Thu, 27 Aug 2015 19:40:46 -0700
Subject: [PATCH 049/188] remoteproc: r5_remoteproc: unset amba reset when
 release RPU reset

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

This issue is found when using the latest FSBL since it is no longer
releasing the AMBA reset.

When releasing RPU from reset, we should also release the AMBA reset

Signed-off-by: Wendy Liang <jliang@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit d2ccd469f8aa53002343c9fcdb4efc30a9719b0a)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/remoteproc/zynqmp_r5_remoteproc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/remoteproc/zynqmp_r5_remoteproc.c b/drivers/remoteproc/zynqmp_r5_remoteproc.c
index c49bd98..b736606 100644
--- a/drivers/remoteproc/zynqmp_r5_remoteproc.c
+++ b/drivers/remoteproc/zynqmp_r5_remoteproc.c
@@ -55,6 +55,7 @@
 #define CPU_R5_CTRL_OFFSET	0x00000090 /* RPU Global Control*/
 #define RST_LPD_TOP_OFFSET	0x0000023C /* LPD block */
 #define RPU0_RESET_BIT		BIT(0) /* RPU CPU0 reset bit */
+#define RPU_AMBA_RST_MASK	BIT(2) /* RPU AMBA reset bit */
 
 /* IPI reg offsets */
 #define TRIG_OFFSET		0x00000000
@@ -220,7 +221,7 @@ static void hw_r5_reset(struct zynqmp_r5_rproc_pdata *pdata,
 	if (do_reset)
 		tmp |= (RPU0_RESET_BIT << pdata->rpu_id);
 	else
-		tmp &= ~(RPU0_RESET_BIT << pdata->rpu_id);
+		tmp &= ~((RPU0_RESET_BIT << pdata->rpu_id) | RPU_AMBA_RST_MASK);
 	reg_write(pdata->crl_apb_base, RST_LPD_TOP_OFFSET, tmp);
 }
 
-- 
2.0.2

